devtools/x/src/cargo/selected_package.rs

```diff
@@ -70,7 +74,28 @@ impl SelectedPackageArgs {
             );
         }
 
-        Ok(SelectedPackages::new(includes))
+        let mut ret = SelectedPackages::new(includes);
+
+        if !self.exclude.is_empty() {
```

Comment:
Should this be a reusable function somewhere?   

xctx.core.fitlerpackages(Vec<_> ) -> (Vec<_>, Vec<_>)
and applied to both includes and excludes? 

Just food for thought... looking for coupling.

Reply:
Hmm I feel like `SelectedPackages` is the reusable type, do you think there can be further abstractions introduced?

Reply:
I suppose so 🤷  thought the function size gets to be a smell for me.   The validation logic that is performed by these lines seems to be a separate concern from the package selection.   One is input validation, the other is transformation of that input in to an exploded list of targets.

Reply:
Yeah agreed, will factor out.

Reply:
Done, called it `partition_workspace_names`.